src/Core/src/Response.php

```diff
@@ -61,10 +62,16 @@ class Response
      */
     private $didThrow = false;
 
-    public function __construct(ResponseInterface $response, HttpClientInterface $httpClient)
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    public function __construct(ResponseInterface $response, HttpClientInterface $httpClient, LoggerInterface $logger)
     {
         $this->httpResponse = $response;
         $this->httpClient = $httpClient;
+        $this->logger = $logger;
```

Comment:
what's about `$ths->logger = $logger ?? new NullLogger()` and remove code from Tests classes

Reply:
I was considering that, but the only win is that test classes looks a bit cleaner. 

There is no point in doing that check on runtime.